components/fxa-client/src/internal/state_manager.rs

```diff
@@ -175,7 +175,9 @@ impl StateManager {
             self.persisted_state.scoped_keys.insert(scope, key);
         }
         self.persisted_state.refresh_token = Some(refresh_token);
-        self.persisted_state.session_token = session_token;
+        if let Some(session_token) = session_token {
+            self.persisted_state.session_token = Some(session_token)
+        }
```

Comment:
Can you add a comment describing how this interacts with the session token from `set_user_data()`?  It seems like it would override it if set, but a comment would be great.

Also, maybe it would be more clear if the code was `self.persisted_state.session_token = session_token.or(self.persisted_state.session_token)`.  I don't feel strongly about that though.

Reply:
The new version of this is super clear thanks!